module/data/region-behavior/difficult-terrain.mjs

```diff
@@ -0,0 +1,89 @@
+const { NumberField, SetField, StringField } = foundry.data.fields;
+
+/**
+ * The data model for a region behavior that represents an area of difficult terrain.
+ */
+export default class DifficultTerrainRegionBehaviorType extends foundry.data.regionBehaviors.RegionBehaviorType {
+
+  /** @override */
+  static LOCALIZATION_PREFIXES = ["DND5E.REGIONBEHAVIORS.DIFFICULTTERRAIN"];
+
+  /* ---------------------------------------- */
+
+  /** @override */
+  static defineSchema() {
+    return {
+      types: new SetField(new StringField({ choices: () => CONFIG.DND5E.difficultTerrainTypes })),
+      ignoredDispositions: new SetField(new NumberField({
+        choices: () => {
+          const dispositions = { ...foundry.applications.sheets.TokenConfig.TOKEN_DISPOSITIONS };
+          delete dispositions[CONST.TOKEN_DISPOSITIONS.SECRET];
+          return dispositions;
+        }
+      }))
+    };
```

Comment:
This would be a bit better for performance: a `choices` function is called every validation.

```suggestion
    const dispositions = { ...foundry.applications.sheets.TokenConfig.TOKEN_DISPOSITIONS };
    delete dispositions[CONST.TOKEN_DISPOSITIONS.SECRET];
    return {
      types: new SetField(new StringField({ choices: () => CONFIG.DND5E.difficultTerrainTypes })),
      ignoredDispositions: new SetField(new NumberField({ choices: dispositions }))
    };
```